### PR TITLE
[FLINK-10569] Refactor CoLocationConstraintTest and remove Instance u…

### PR DESCRIPTION
**Pull request by TisonKun**
…sage

## What is the purpose of the change

Get rid of unnecessary try/catch block. `CoLocationConstraint` now works with `setSlotRequestId` and `lockLocation(TaskManagerLocation)`. Test the new logic.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector:(no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

cc @tillrohrmann

### CODE REVIEW

**Comment by zentol** [comment]
please move the try/catch block removal into a separate commit.

**Comment by TisonKun** [comment]
Address comments.

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/scheduler/CoLocationConstraintTest.java [hunk 1]
```diff
@@ -64,39 +65,25 @@ public void testCreateConstraints() {
...
-		SharedSlot slot2_2 = instance2.allocateSharedSlot(assignment);
+		assertThat(constraint.getSlotRequestId(), is(nullValue()));
+		assertThat(constraint.isAssigned(), is(false));
 
-		// constraint is still completely unassigned
-		assertFalse(constraint.isAssigned());
-		assertFalse(constraint.isAssignedAndAlive());
+		SlotRequestId slotRequestId1 = new SlotRequestId();
+		constraint.setSlotRequestId(slotRequestId1);
+		assertThat(constraint.getSlotRequestId(), is(slotRequestId1));
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/scheduler/CoLocationConstraintTest.java]: given that these are plain getters & setters I believe we can omit these assertions

Comment (TisonKun) [reply flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/scheduler/CoLocationConstraintTest.java]: resolved.

**Pull request merged**